include/rmm/cuda_stream_pool.hpp

```diff
@@ -61,6 +62,33 @@ class cuda_stream_pool {
     return streams_[(next_stream++) % streams_.size()].view();
   }
 
+  /**
+   * @brief Get a `cuda_stream_view` of the stream at stream_id index the pool.
```

Comment:
Instead of treating `stream_index` as a literal index into the internal storage (an implementation detail) the `stream_index` should be renamed to `stream_id` with the guarantee that equivalent values of `stream_id` will always return a `stream_view` to the same underlying stream.

Reply:
Sounds good to me. Added in 898cf5d